.github/workflows/run-tests.yaml

```diff
@@ -0,0 +1,72 @@
+name: Run Tests
```

Comment:
Maybe worth changing both worflow name and filename to something more unique ? run tests can also be nightly
maybe run pr tests, or anything else which would be a bit more clear?